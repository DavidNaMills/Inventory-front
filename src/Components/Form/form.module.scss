@import '../../styles/variables/sizing.module.scss';
@import '../../styles/variables/colors.module.scss';

@mixin form {
    -webkit-backdrop-filter: blur(5px);
    backdrop-filter: blur(5px);
    // background-color: rgba(242, 242, 242, 0.5);

    margin-bottom: $largeSpacing;
    margin-top: $mediumSpacing;
    z-index: 1;
}



.form{
    border-radius: 3px;
    padding: 15px;
    
    
    -webkit-box-shadow: $blur $shadow-blue;
    -moz-box-shadow: $blur $shadow-blue;
    box-shadow: $blur $shadow-blue;

    &__buttonSpacing{
        &_large{
            margin-top: $mediumSpacing;
        }
        &_medium{
            margin-top: $extraSmallSpacing;
        }
    }
}

.blur{
    &__normal{
        width: 40%;
        @include form;
    }
    
    &_mobile{
        width: 80%;
        @include form;
    }
}




// @import '../../styles/variables/sizing.module.scss';
// @import '../../styles/variables/colors.module.scss';

// @mixin form {
//     border-radius: 3px;
//     padding: 15px;
//     margin-bottom: $largeSpacing;
//     margin-top: $mediumSpacing;
    
//     -webkit-box-shadow: $blur $shadow-blue;
//     -moz-box-shadow: $blur $shadow-blue;
//     box-shadow: $blur $shadow-blue;


// }


// .form{
//     &__normal{
//         width: 40%;
//         @include form;
//     }
    
//     &_mobile{
//         width: 80%;
//         @include form;
//     }

//     &__buttonSpacing{
//         &_large{
//             margin-top: $mediumSpacing;
//         }
//         &_medium{
//             margin-top: $extraSmallSpacing;
//         }
//     }
// }

// .blur{
//     // width:100%;
//     -webkit-backdrop-filter: blur(10px);
//     backdrop-filter: blur(5px);
//     background-color: rgba(255, 255, 255, 0.5); 
// }