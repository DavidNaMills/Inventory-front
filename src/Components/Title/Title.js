import React from 'react';
import classes from './title.module.scss';


const Title = (props) => {
    const { size = null, message = '', center = false } = props;
    let element = null;

    switch (size) {
        case 'h1':
            element = <h1 className={[classes.title_h1, center ? classes.title__center : null].join(' ')} >{message}</h1>
            break;
        case 'h2':
            element = <h2 className={[classes.title_h2, center ? classes.title__center : null].join(' ')}>{message}</h2>
            break;
        case 'h3':
            element = <h3 className={[classes.title_h3, center ? classes.title__center : null].join(' ')}>{message}</h3>
            break;
        case 'h4':
            element = <h4 className={[classes.title_h4, center ? classes.title__center : null].join(' ')}>{message}</h4>
            break;
        case 'h5':
            element = <h5 className={[classes.title_h5, center ? classes.title__center : null].join(' ')}>{message}</h5>
            break;
        case 'h6':
            element = <h6 className={[classes.title_h6, center ? classes.title__center : null].join(' ')}>{message}</h6>
            break;
        default:
            element = <h2 className={[classes.title_h2, center ? classes.title__center : null].join(' ')}>{message}</h2>
            break;
    };

    return (
        <div>
            {element}
        </div>
    )
}

export default React.memo(Title);