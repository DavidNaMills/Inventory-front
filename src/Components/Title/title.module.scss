@import '../../styles/variables/colors.module.scss';
@import '../../styles/fonts/fontSizes.module.scss';
@import '../../styles/mixins/title.module.scss';

.title {
    @include setTitle($h2, $title-color);
    
    &_h1{
        @include setTitle($h1, $title-color);
    };

    &_h2{
        @include setTitle($h2, $title-color);
    };
    
    &_h3{
        @include setTitle($h3, $title-color);
    }

    &_h4{
        @include setTitle($h4, $title-color);
    }

    &_h5{
        @include setTitle($h5, $title-color);
    }

    &_h6{
        @include setTitle($h6, $title-color);
    }

    &__center{
        text-align: center;
    }


}