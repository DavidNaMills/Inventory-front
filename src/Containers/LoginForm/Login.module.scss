.image_container {
    position: fixed;
    left: 0;
    right: 0;
    z-index: 1;
    display: block;

    
    @media (max-width: 1100px){
        background: url('/images/logo-small.jpg') no-repeat center center fixed; 
    }
    @media (min-width: 1101px){
        background: url('/images/logo.jpg') no-repeat center center fixed; 
    }

    width: 100%;
    height: 100%;

    backdrop-filter: grayscale(1);
}

.overlay{
    position: fixed;
    left: 0;
    right: 0;
    z-index: 9996;
    height: 80%;
    width: 100%;
    filter: none;
}

.login{
    height: 100%;
    width: 100%;
    display: flex;
    flex-direction: column;
    justify-content: center;
    align-items: center;
}