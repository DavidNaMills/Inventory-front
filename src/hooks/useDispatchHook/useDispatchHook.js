import { useDispatch } from 'react-redux';

import { userLogin, userLogout, changeCurrentLocation } from '../../store/userStore/userActions';
import { startInitTypes, addNewProductType, addNewLocation, removeLocationType } from '../../store/typesStore/typesStoreActions';
import {createMessage} from '../../store/messagingStore/messagingActions';

const useDispatchHook = () => {
    const dispatch = useDispatch();

    const userLoginDispatch = (userData) => dispatch(userLogin(userData));
    const userLogoutDispatch = () => dispatch(userLogout());
    const changeCurrentLocationDispatch = (id) => dispatch(changeCurrentLocation(id));

    const initTypesDispatch = () => dispatch(startInitTypes());
    const addNewProductDispatch = (newType) => dispatch(addNewProductType(newType));
    const addNewLocationDispatch = (newLocation) => dispatch(addNewLocation(newLocation));
    const removeLocationDispatch = (id) => dispatch(removeLocationType(id));

    const createMessageDispatch = (data) => dispatch(createMessage(data));

    return {
        userLoginDispatch,
        userLogoutDispatch,
        changeCurrentLocationDispatch,

        initTypesDispatch,
        addNewProductDispatch,
        addNewLocationDispatch,
        removeLocationDispatch,

        createMessageDispatch
    }
}

export default useDispatchHook;