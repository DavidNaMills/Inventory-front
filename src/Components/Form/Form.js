import React from 'react';
import classes from './form.module.scss';
import { isMobileOnly } from 'react-device-detect';
import { useLocale } from '../../Context/LocaleContext';
import FormFactory from './Factory/FormFactory';
import Button from '../Button/Button';
import Title from '../Title/Title';
import PublishIcon from '@material-ui/icons/Publish';
import ClearIcon from '@material-ui/icons/Clear';

import objToArray from '../../helpers/objToArray/objToArray';


const Form = (props) => {
    const locale = useLocale();
    const {
        displayFields,
        user,
        fullForm,
        controlForm,
        updating,
        withTitle = true,

        submitCb = null,
        clearCb = null
    } = props;
    // isMobileOnly ? classes.form_mobile : classes.form__normal}
    return (
        <div className={isMobileOnly ? classes.blur_mobile : classes.blur__normal}>
            <div className={classes.form}>
                {withTitle && <Title message={updating ? `${locale.updating} ${updating}` : locale.creating} size='h3' />}
                <form onSubmit={submitCb}>
                    {
                        objToArray(fullForm).map((x, y) =>
                            <FormFactory
                                key={y}
                                config={x}
                                role={user.role}
                                displayFields={displayFields}
                                onChange={controlForm}
                            />
                        )
                    }
                    <div className={classes.form__buttonSpacing_large}>
                        <Button config={{ type: 'submit' }} type="submit" label={updating ? locale.update : locale.submit} Icon={PublishIcon} full />
                    </div>
                </form>
                {clearCb && <div className={classes.form__buttonSpacing_medium}>
                    <Button onClick={clearCb} label={locale.clear} type='clear' Icon={ClearIcon} full />
                </div>}
            </div>
        </div>
    )
}

export default Form;