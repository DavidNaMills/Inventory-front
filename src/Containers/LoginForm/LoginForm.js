import React, { useEffect } from 'react';
import classes from './Login.module.scss';
import Form from '../../Components/Form/Form';
import { loginConfig } from '../../configs/formConfigs/loginFormConfig';

import axios from '../../Axios/axiosInstance';
import useAxios from '../../hooks/useAxios/useAxios';
import useForm from '../../hooks/useForm/useForm';
import useDispatchHook from '../../hooks/useDispatchHook/useDispatchHook';
import { useLocale } from '../../Context/LocaleContext';

import Title from '../../Components/Title/Title';
import Spinner from '../../Components/Spinner/Spinner';

const tempUserConfig = {
    role: 99,
    formName: {
        login: {
            create: {
                phone: true,
                password: true
            }
        }
    }
}

const FORM_NAME = 'login';

const LoginForm = () => {
    const locale = useLocale();
   

    const { userLoginDispatch, initTypesDispatch } = useDispatchHook();
    const { fetchState, makeCall } = useAxios(axios);

    const {
        fullForm,
        formRef,
        buildFormObject,
        controlForm,
        validateFull
    } = useForm({ formConfig: loginConfig(locale.errors) });




    // useEffect(()=>{
    //     makeCall({
    //         url: '/login',
    //         method: 'POST',
    //         data: {phone: '13103718079', password: 'david_mills'},
    //         callBack: loginCallback
    //     })
    // }, []);


    const loginCallback = (res) => {
        if (res.staff && res.token) {
            initTypesDispatch();
            userLoginDispatch(res);
        }
    };


    const submitForm = (e) => {
        e.preventDefault();

        const valid = validateFull();
        if (valid) {
            const all = buildFormObject();

            makeCall({
                url: '/login',
                method: 'POST',
                data: all,
                callBack: loginCallback
            })
        }
    }

    return (
        <React.Fragment>
            <div className={classes.image_container}></div>
            <div className={classes.overlay}>
                <div className={classes.login}>
                    
                {fetchState.isLoading && <div><Spinner/></div>}
                <React.Fragment>
                    <Title size={'h1'} message={locale[FORM_NAME]} />
                </React.Fragment>
                <React.Fragment>
                <Form
                    controlForm={controlForm}
                    title='Login'
                    displayFields={tempUserConfig.formName.login.create}
                    formName={FORM_NAME}
                    user={tempUserConfig}
                    fullForm={fullForm}
                    formRef={formRef}
                    submitCb={submitForm}
                    withTitle={false}
                />
                </React.Fragment>
                </div>
            </div>
        </React.Fragment>
    )
}

export default LoginForm;