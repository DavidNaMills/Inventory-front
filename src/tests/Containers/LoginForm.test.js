import React from 'react';
import '../testHelpers/context';
import renderer from 'react-test-renderer';
import { shallow, mount } from 'enzyme';

import LoginForm from '../../Containers/LoginForm/LoginForm';
import Title from '../../Components/Title/Title';
import Form from '../../Components/Form/Form';

jest.mock('../../hooks/useDispatchHook/useDispatchHook');
import useDispatchHook from '../../hooks/useDispatchHook/useDispatchHook';

const mockUserLoginDispatch = jest.fn();
const mockInitTypesDispatch = jest.fn();

describe('<LoginForm/> test suite', () => {
    beforeEach(()=>{
        useDispatchHook.mockImplementation(()=>({
            userLoginDispatch: mockUserLoginDispatch,
            initTypesDispatch: mockInitTypesDispatch
        }));
    });

    afterEach(()=>{
        jest.clearAllMocks();
    });

    it('matches snapshot', () => { 
        const comp = renderer.create(<LoginForm />);
        const app = comp.toJSON();
        expect(app).toMatchSnapshot();
    });
    
    it('exists', () => { 
        const wrapper = mount(<LoginForm />);
        expect(wrapper.exists).toBeTruthy();
    });
    
    it('renders a <Title/> components', () => { 
        const wrapper = shallow(<LoginForm />);
        const title = wrapper.find(Title);

        expect(title.length).toBe(1);
        expect(title.props()).toHaveProperty('size', 'h1');
        expect(title.props()).toHaveProperty('message', 'login');
    });
    
    it('renders a <Form/> components', () => { 
        const wrapper = mount(<LoginForm />);
        const tForm = wrapper.find(Form);

        expect(tForm.length).toBe(1);
        expect(tForm.props()).toHaveProperty('title', 'Login');
        expect(tForm.props()).toHaveProperty('formName', 'login');
    });
});