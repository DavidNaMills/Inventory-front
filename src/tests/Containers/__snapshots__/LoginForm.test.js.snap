// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LoginForm/> test suite matches snapshot 1`] = `
Array [
  <div
    className="image_container"
  />,
  <div
    className="overlay"
  >
    <div
      className="login"
    >
      <div>
        <h1
          className="title_h1 "
        >
          Login
        </h1>
      </div>
      <div
        className="blur__normal"
      >
        <div
          className="form"
        >
          <form
            onSubmit={[Function]}
          >
            <div
              className="form__buttonSpacing_large"
            >
              <button
                className="button button__submit button_full"
                disabled={false}
                style={Object {}}
                type="submit"
              >
                <p>
                  Submit
                </p>
                <svg
                  aria-hidden="true"
                  className="MuiSvgIcon-root"
                  focusable="false"
                  role="presentation"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M5 4v2h14V4H5zm0 10h4v6h6v-6h4l-7-7-7 7z"
                  />
                </svg>
              </button>
            </div>
          </form>
        </div>
      </div>
    </div>
  </div>,
]
`;
